### PR TITLE
Basic support for autocompletion.

### PR DESCRIPTION
**Pull request by olafurpg**
- requires manual `project/config:scalametaEnableCompletions` step in
  sbt to setup the compiler
- Each new/changed/removed file in the classpath requires scalametaEnableCompletions
  to be run again
- requires entire project to compile before setting up
- works for multi-module/multi-config/multi-sourcefile projects
- currently only works for scala 2.12.3, it's maybe possible
  to support a bigger variety of scala compiler versions using reflection.
- most probably leaks memory, we need to properly clean up resources
  from old compilers when constructing new compilers.
- most probably very slow for large projects, we may want to use the
  same tricks as scalafiddle-core to store classpath in-memory.

In summary, this is a promising start but there is a lot of work to be
done. The recommended workflow is the following

1. compile everything
2. run `> project/config:scalametaEnableCompletions` for the
project+configuration where you want to work on a single file.
  For example, `server/test:scalametaEnableCompletions` to work
  on the test suite of your server project.
3. Edit that file, enjoy completions as long as you edit that file.
4. Repeat process whenever you
  - switch project/configuration OR
  - create/delete a *.scala file OR
  - change a *.scala file from the classpath of project/configuration

usertest

### CODE REVIEW

File: language-server/src/main/scala/scala/meta/languageserver/Compiler.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interactive.Global
+import scala.tools.nsc.reporters.StoreReporter
+import com.typesafe.scalalogging.LazyLogging
+import langserver.core.Connection
+import langserver.messages.MessageType
+import monix.reactive.Observable
+import org.langmeta.io.AbsolutePath
+
+case class CompilerConfig(
```
Comment (jvican) [inline language-server/src/main/scala/scala/meta/languageserver/Compiler.scala]: Don't you need the scala version here?

Comment (olafurpg) [reply language-server/src/main/scala/scala/meta/languageserver/Compiler.scala]: We will need to classload the compiler for it to work for more scala versions, that's the next step. Currently it's hardcoded for 2.12.3. We will likely only support the 2.11/2.12 releases supported by semanticdb, in order to provide scalafix linting as you type alongside completions.
